tools/lxdclient/client.go

```diff
@@ -251,6 +242,46 @@ func newRawClient(remote Remote) (*lxd.Client, error) {
 	return client, nil
 }
 
+// Generate host string from remote.ID and remote.Host
+func generateHostString(id string, host string) (string) {
+	if host != "" {
+		is_localhost, _ := isLocalhost(host)
+		if is_localhost {
+			return "unix://" + lxdshared.VarPath("unix.socket")
+		} else {
+			return net.JoinHostPort(host, lxdshared.DefaultPort)
+		}
+	} else if id == remoteIDForLocal && host == "" {
+		return "unix://" + lxdshared.VarPath("unix.socket")
+	} else if !strings.HasPrefix(host, "unix://") {
+		_, _, err := net.SplitHostPort(host)
+		if err != nil {
+			// There is no port here
+			return net.JoinHostPort(host, lxdshared.DefaultPort)
+		}
+	}
+	return host
+}
+
+// Check if the remote host is pointing to a local ip address configured in
+// our interfaces.
+func isLocalhost(host string) (bool, error) {
```

Comment:
if we plan to put the hostname instead of the ip address in the configuration of the cloud yes, but I guess we can live with this for now, and add the name resolution later.